airflow/decorators/base.py

```diff
@@ -101,22 +102,22 @@ class DecoratedOperator(BaseOperator):
     :type kwargs_to_upstream: dict
     """
 
-    template_fields = ('op_args', 'op_kwargs')
+    template_fields: Tuple[str, ...] = ('op_args', 'op_kwargs')
     template_fields_renderers = {"op_args": "py", "op_kwargs": "py"}
 
     # since we won't mutate the arguments, we should just do the shallow copy
     # there are some cases we can't deepcopy the objects (e.g protobuf).
-    shallow_copy_attrs = ('python_callable',)
+    shallow_copy_attrs: Tuple[str, ...] = ('python_callable',)
 
     def __init__(
         self,
         *,
         python_callable: Callable,
         task_id: str,
-        op_args: Tuple[Any],
+        op_args: List,
```

Comment:
and this. Maybe even more. I get a feeling we should annotate BaseOperator first before any other operators.

Reply:
This one is should share the type annotation with whatever PythonOperator `op_args` has.

Reply:
I’m changing this one to `Collection[Any]` in #20376.

Reply:
Ah, I'll take a look at that PR and make sure this PR aligns. Thanks for the heads up.

Reply:
@uranusjr I don't see the `Collection[Any]` typing in that PR. Should I reference another to align with?